kcbq-connector/src/test/java/com/wepay/kafka/connect/bigquery/BigQuerySinkTaskTest.java

```diff
@@ -297,70 +297,40 @@ public void testPutWhenPartitioningOnMessageTimeWhenNoTimestampType() {
     Storage storage = mock(Storage.class);
-
+    String error = "Cannot add required fields to an existing schema.";
     SinkTaskContext sinkTaskContext = mock(SinkTaskContext.class);
-    InsertAllResponse insertAllResponse = mock(InsertAllResponse.class);
-    when(bigQuery.insertAll(any())).thenReturn(insertAllResponse);
-    when(insertAllResponse.hasErrors()).thenReturn(true);
-    when(insertAllResponse.getInsertErrors()).thenReturn(Collections.singletonMap(
-       0L, Collections.singletonList(new BigQueryError("no such field", "us-central1", ""))));
+    when(bigQuery.insertAll(any()))
+        .thenThrow(
+            new BigQueryException(400, error, new BigQueryError("invalid", "global", error)));
 
     SchemaRetriever schemaRetriever = mock(SchemaRetriever.class);
     SchemaManager schemaManager = mock(SchemaManager.class);
 
-    BigQuerySinkTask testTask = new BigQuerySinkTask(bigQuery, schemaRetriever, storage, schemaManager);
+    BigQuerySinkTask testTask =
+        new BigQuerySinkTask(bigQuery, schemaRetriever, storage, schemaManager);
     testTask.initialize(sinkTaskContext);
     testTask.start(properties);
 
     testTask.put(Collections.singletonList(spoofSinkRecord(topic)));
-    while (true) {
-      Thread.sleep(100);
-      testTask.put(Collections.emptyList());
-    }
-  }
-
-  // Since any exception thrown during flush causes Kafka Connect to not commit the offsets for any
-  // records sent to the task since the last flush. The task should fail, and next flush should
-  // also throw an error.
-  @Test(expected = BigQueryConnectException.class)
-  public void testFlushException() {
-    final String dataset = "scratch";
-    final String topic = "test_topic";
-
-    Map<String, String> properties = propertiesFactory.getProperties();
-    properties.put(BigQuerySinkConfig.TOPICS_CONFIG, topic);
-    properties.put(BigQuerySinkConfig.DATASETS_CONFIG, String.format(".*=%s", dataset));
-
-    BigQuery bigQuery = mock(BigQuery.class);
-    Storage storage = mock(Storage.class);
-    when(bigQuery.insertAll(any(InsertAllRequest.class)))
-        .thenThrow(new RuntimeException("This is a test"));
-
-    SchemaRetriever schemaRetriever = mock(SchemaRetriever.class);
-    SchemaManager schemaManager = mock(SchemaManager.class);
-
-    SinkTaskContext sinkTaskContext = mock(SinkTaskContext.class);
-    BigQuerySinkTask testTask = new BigQuerySinkTask(bigQuery, schemaRetriever, storage, schemaManager);
-    testTask.initialize(sinkTaskContext);
-    testTask.start(properties);
-
     try {
-      testTask.put(Collections.singletonList(spoofSinkRecord(topic)));
-      testTask.flush(Collections.emptyMap());
-      fail("An exception should have been thrown by now");
-    } catch (BigQueryConnectException err) {
-      testTask.flush(Collections.emptyMap());
-      verify(bigQuery, times(1)).insertAll(any(InsertAllRequest.class));
+      while (true) {
+        Thread.sleep(100);
```

Comment:
What is this?

Reply:
Wait a bit to do the second `put()`, or the task won't fail.